From: Michael Froman <mfroman@mozilla.com>
Date: Mon, 19 Jan 2026 14:35:01 -0600
Subject: (tmp-cherry-pick) Revert "Reland "Passing video codec settings to
 lower layers"" (8e4a86ee69)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This reverts commit e1d6be23f9c0ed0187603e5348d525060f3869d2.

Reason for revert:
Caused downstream test to start flaking, see https://crbug.com/362277533#comment64 for more details.

Original change's description:
> Reland "Passing video codec settings to lower layers"
>
> This is a reland of
> https://webrtc-review.googlesource.com/c/src/+/412641
>
> In the previous implementation, when the codec was switched in
> RequestEncoderSwitch(), there was an issue where video_format still
> referred to the old codec.
>
> By updating rtp_parameters_ in WebRtcVideoSendStream::SetCodec(),
> video_format is now updated to the correct codec.
>
> Original change's description:
> > Passing video codec settings to lower layers
> >
> > In the CL at https://webrtc-review.googlesource.com/c/src/+/387320
> > I added `SimulcastStream::video_format`, but I forgot to set this value
> > in the upper layer.
> >
> > Bug: webrtc:362277533
> > Change-Id: I16fa78e99c1998d3fbc7aca03a83e74fa8278d7d
> > Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/412641
> > Commit-Queue: Henrik Boström <hbos@webrtc.org>
> > Reviewed-by: Erik Språng <sprang@webrtc.org>
> > Reviewed-by: Harald Alvestrand <hta@webrtc.org>
> > Reviewed-by: Henrik Boström <hbos@webrtc.org>
> > Cr-Commit-Position: refs/heads/main@{#46020}
>
> Bug: webrtc:362277533
> Change-Id: I88f674f4ec98364d7c6edb3d4566748815b19084
> Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/427640
> Reviewed-by: Henrik Boström <hbos@webrtc.org>
> Reviewed-by: Harald Alvestrand <hta@webrtc.org>
> Commit-Queue: Henrik Boström <hbos@webrtc.org>
> Cr-Commit-Position: refs/heads/main@{#46304}

Bug: webrtc:362277533
Change-Id: Iea4cd47812ec7a2dc2e71ec1abba5ae6e7f23308
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/429180
Reviewed-by: Harald Alvestrand <hta@webrtc.org>
Bot-Commit: Rubber Stamper <rubber-stamper@appspot.gserviceaccount.com>
Auto-Submit: Henrik Boström <hbos@webrtc.org>
Commit-Queue: Henrik Boström <hbos@webrtc.org>
Commit-Queue: Harald Alvestrand <hta@webrtc.org>
Cr-Commit-Position: refs/heads/main@{#46335}
---
 media/engine/webrtc_video_engine.cc          |  20 +--
 media/engine/webrtc_video_engine_unittest.cc | 121 -------------------
 2 files changed, 3 insertions(+), 138 deletions(-)

diff --git a/media/engine/webrtc_video_engine.cc b/media/engine/webrtc_video_engine.cc
index 6a269c8545..dd1539581f 100644
--- a/media/engine/webrtc_video_engine.cc
+++ b/media/engine/webrtc_video_engine.cc
@@ -1977,6 +1977,9 @@ void WebRtcVideoSendChannel::WebRtcVideoSendStream::SetCodec(
   FallbackToDefaultScalabilityModeIfNotSupported(
       codec_settings.codec, parameters_.config, rtp_parameters_.encodings);
 
+  parameters_.encoder_config = CreateVideoEncoderConfig(codec_settings.codec);
+  RTC_DCHECK_GT(parameters_.encoder_config.number_of_streams, 0);
+
   parameters_.config.rtp.payload_name = codec_settings.codec.name;
   parameters_.config.rtp.payload_type = codec_settings.codec.id;
   parameters_.config.rtp.raw_payload =
@@ -2044,18 +2047,6 @@ void WebRtcVideoSendChannel::WebRtcVideoSendStream::SetCodec(
     }
   }
 
-  // Reset all encoding codecs if no codec settings list is provided.
-  if (codec_settings_list.empty()) {
-    for (size_t i = 0; i < rtp_parameters_.encodings.size(); i++) {
-      rtp_parameters_.encodings[i].codec.reset();
-    }
-  }
-
-  // CreateVideoEncoderConfig depends on the contents of rtp_parameters_, so
-  // rtp_parameters_ must be updated before calling it.
-  parameters_.encoder_config = CreateVideoEncoderConfig(codec_settings.codec);
-  RTC_DCHECK_GT(parameters_.encoder_config.number_of_streams, 0);
-
   parameters_.codec_settings_list = codec_settings_list;
 
   // TODO(bugs.webrtc.org/8830): Avoid recreation, it should be enough to call
@@ -2364,11 +2355,6 @@ WebRtcVideoSendChannel::WebRtcVideoSendStream::CreateVideoEncoderConfig(
       encoder_config.simulcast_layers[i].num_temporal_layers =
           *rtp_parameters_.encodings[i].num_temporal_layers;
     }
-    if (rtp_parameters_.encodings[i].codec) {
-      encoder_config.simulcast_layers[i].video_format =
-          SdpVideoFormat(rtp_parameters_.encodings[i].codec->name,
-                         rtp_parameters_.encodings[i].codec->parameters);
-    }
     encoder_config.simulcast_layers[i].scale_resolution_down_to =
         rtp_parameters_.encodings[i].scale_resolution_down_to;
   }
diff --git a/media/engine/webrtc_video_engine_unittest.cc b/media/engine/webrtc_video_engine_unittest.cc
index 85defbcd3d..e9f93698e2 100644
--- a/media/engine/webrtc_video_engine_unittest.cc
+++ b/media/engine/webrtc_video_engine_unittest.cc
@@ -59,7 +59,6 @@
 #include "api/video/video_codec_constants.h"
 #include "api/video/video_codec_type.h"
 #include "api/video/video_content_type.h"
-#include "api/video/video_frame.h"
 #include "api/video/video_frame_buffer.h"
 #include "api/video/video_rotation.h"
 #include "api/video/video_sink_interface.h"
@@ -9050,55 +9049,6 @@ TEST_F(WebRtcVideoChannelTest,
   EXPECT_TRUE(send_channel_->SetVideoSend(last_ssrc_, nullptr, nullptr));
 }
 
-TEST_F(WebRtcVideoChannelTest, RequestEncoderSwitchClearAllCodecs) {
-  StreamParams sp = CreateSimStreamParams("cname", {123, 456, 789});
-
-  std::vector<RidDescription> rid_descriptions;
-  rid_descriptions.emplace_back("f", RidDirection::kSend);
-  rid_descriptions.emplace_back("h", RidDirection::kSend);
-  rid_descriptions.emplace_back("q", RidDirection::kSend);
-  sp.set_rids(rid_descriptions);
-
-  ASSERT_TRUE(send_channel_->AddSendStream(sp));
-
-  Codec vp8 = GetEngineCodec("VP8");
-  Codec vp9 = GetEngineCodec("VP9");
-
-  // VP9 is set initially for all simulcast layers.
-  RtpParameters rtp_parameters =
-      send_channel_->GetRtpSendParameters(last_ssrc_);
-  EXPECT_EQ(3UL, rtp_parameters.encodings.size());
-  rtp_parameters.encodings[0].codec = vp9.ToCodecParameters();
-  rtp_parameters.encodings[1].codec = vp9.ToCodecParameters();
-  rtp_parameters.encodings[2].codec = vp9.ToCodecParameters();
-  EXPECT_TRUE(
-      send_channel_->SetRtpSendParameters(last_ssrc_, rtp_parameters).ok());
-
-  // Switch to VP8
-  SendImpl()->RequestEncoderSwitch(SdpVideoFormat::VP8(),
-                                   /*allow_default_fallback=*/true);
-
-  // Verify that VP8 is set for all simulcast layers and RtpStreamConfig is
-  // cleared.
-  const auto& streams = fake_call_->GetVideoSendStreams();
-  ASSERT_EQ(1u, streams.size());
-  auto stream = streams[0];
-  ASSERT_NE(stream, nullptr);
-  const auto& config = stream->GetConfig();
-  EXPECT_EQ(vp8.name, config.rtp.payload_name);
-  EXPECT_EQ(vp8.id, config.rtp.payload_type);
-  EXPECT_THAT(config.rtp.stream_configs, SizeIs(0));
-
-  VideoEncoderConfig encoder_config = stream->GetEncoderConfig().Copy();
-  EXPECT_EQ(vp8.name, config.rtp.payload_name);
-  EXPECT_EQ(vp8.id, config.rtp.payload_type);
-  EXPECT_EQ(vp8.name, encoder_config.video_format.name);
-  ASSERT_THAT(encoder_config.simulcast_layers, SizeIs(3));
-  EXPECT_EQ(std::nullopt, encoder_config.simulcast_layers[0].video_format);
-  EXPECT_EQ(std::nullopt, encoder_config.simulcast_layers[1].video_format);
-  EXPECT_EQ(std::nullopt, encoder_config.simulcast_layers[2].video_format);
-}
-
 class WebRtcVideoChannelWithMixedCodecSimulcastTest
     : public WebRtcVideoChannelTest {
  public:
@@ -9152,77 +9102,6 @@ TEST_F(WebRtcVideoChannelWithMixedCodecSimulcastTest,
   EXPECT_EQ(config.rtp.stream_configs[2].payload_type, vp9.id);
 }
 
-// Verify that RtpEncodingParameters::codec propagates to
-// VideoEncoderConfig::video_format.
-TEST_F(WebRtcVideoChannelWithMixedCodecSimulcastTest,
-       CodecSettingsPropagatedToEncoder) {
-  FakeVideoSendStream* stream = SetUpSimulcast(true, /*with_rtx=*/false);
-
-  // Send a full size frame so all simulcast layers are used when reconfiguring.
-  FrameForwarder frame_forwarder;
-  VideoOptions options;
-  EXPECT_TRUE(
-      send_channel_->SetVideoSend(last_ssrc_, &options, &frame_forwarder));
-  send_channel_->SetSend(true);
-  frame_forwarder.IncomingCapturedFrame(frame_source_.GetFrame());
-
-  // Get and set the rtp encoding parameters.
-  RtpParameters parameters = send_channel_->GetRtpSendParameters(last_ssrc_);
-  EXPECT_EQ(kNumSimulcastStreams, parameters.encodings.size());
-
-  // Verify that the initial value is nullopt.
-  VideoEncoderConfig encoder_config = stream->GetEncoderConfig().Copy();
-  EXPECT_EQ(kNumSimulcastStreams, encoder_config.number_of_streams);
-  EXPECT_EQ(kNumSimulcastStreams, encoder_config.simulcast_layers.size());
-  EXPECT_FALSE(encoder_config.simulcast_layers[0].video_format);
-  EXPECT_FALSE(encoder_config.simulcast_layers[1].video_format);
-  EXPECT_FALSE(encoder_config.simulcast_layers[2].video_format);
-
-  // Set single-codec simulcast
-  parameters.encodings[0].codec.emplace(
-      GetEngineCodec("VP8").ToCodecParameters());
-  parameters.encodings[1].codec.emplace(
-      GetEngineCodec("VP8").ToCodecParameters());
-  parameters.encodings[2].codec.emplace(
-      GetEngineCodec("VP8").ToCodecParameters());
-  EXPECT_TRUE(send_channel_->SetRtpSendParameters(last_ssrc_, parameters).ok());
-
-  // Verify that the new value propagated down to the encoder.
-  stream = fake_call_->GetVideoSendStreams()[0];
-  encoder_config = stream->GetEncoderConfig().Copy();
-  EXPECT_EQ(kNumSimulcastStreams, encoder_config.number_of_streams);
-  EXPECT_EQ(kNumSimulcastStreams, encoder_config.simulcast_layers.size());
-  EXPECT_TRUE(encoder_config.simulcast_layers[0].video_format);
-  EXPECT_TRUE(encoder_config.simulcast_layers[1].video_format);
-  EXPECT_TRUE(encoder_config.simulcast_layers[2].video_format);
-  EXPECT_EQ("VP8", encoder_config.simulcast_layers[0].video_format->name);
-  EXPECT_EQ("VP8", encoder_config.simulcast_layers[1].video_format->name);
-  EXPECT_EQ("VP8", encoder_config.simulcast_layers[2].video_format->name);
-
-  // Set mixed-codec simulcast
-  parameters.encodings[0].codec.emplace(
-      GetEngineCodec("VP8").ToCodecParameters());
-  parameters.encodings[1].codec.emplace(
-      GetEngineCodec("VP8").ToCodecParameters());
-  parameters.encodings[2].codec.emplace(
-      GetEngineCodec("VP9").ToCodecParameters());
-  EXPECT_TRUE(send_channel_->SetRtpSendParameters(last_ssrc_, parameters).ok());
-
-  // Verify that the new value propagated down to the encoder.
-  stream = fake_call_->GetVideoSendStreams()[0];
-  encoder_config = stream->GetEncoderConfig().Copy();
-  EXPECT_EQ(kNumSimulcastStreams, encoder_config.number_of_streams);
-  EXPECT_EQ(kNumSimulcastStreams, encoder_config.simulcast_layers.size());
-  EXPECT_TRUE(encoder_config.simulcast_layers[0].video_format);
-  EXPECT_TRUE(encoder_config.simulcast_layers[1].video_format);
-  EXPECT_TRUE(encoder_config.simulcast_layers[2].video_format);
-  EXPECT_EQ("VP8", encoder_config.simulcast_layers[0].video_format->name);
-  EXPECT_EQ("VP8", encoder_config.simulcast_layers[1].video_format->name);
-  EXPECT_EQ("VP9", encoder_config.simulcast_layers[2].video_format->name);
-
-  EXPECT_TRUE(send_channel_->SetVideoSend(last_ssrc_, nullptr, nullptr));
-}
-
 #if RTC_DCHECK_IS_ON && GTEST_HAS_DEATH_TEST && !defined(WEBRTC_ANDROID)
 TEST_F(WebRtcVideoChannelWithMixedCodecSimulcastTest,
        SetMixedCodecSimulcastWithDifferentConfigSettingsSizes) {
